// src/controllers/alumnoController.js
const Alumno = require('../models/alumnoModel');

exports.getAllAlumnos = (req, res) => {
    Alumno.getAll((err, results) => {
        if (err) res.status(500).send(err);
        res.json(results);
    });
};

exports.getAlumnoById = (req, res) => {
    Alumno.getById(req.params.id, (err, results) => {
        if (err) res.status(500).send(err);
        res.json(results);
    });
};

exports.createAlumno = (req, res) => {
    Alumno.create(req.body, (err, results) => {
        if (err) res.status(500).send(err);
        res.json(results);
    });
};

exports.updateAlumno = (req, res) => {
    Alumno.update(req.params.id, req.body, (err, results) => {
        if (err) res.status(500).send(err);
        res.json(results);
    });
};

exports.deleteAlumno = (req, res) => {
    Alumno.delete(req.params.id, (err, results) => {
        if (err) res.status(500).send(err);
        res.json(results);
    });
};

exports.getEstadisticasEdad = (req, res) => {
    Alumno.getEstadisticasEdad((err, results) => {
        if (err) res.status(500).send(err);
        res.json(results);
    });
};
